### DISCUSSION TITLE
Permission to create but not to read. How to get the ID?

### DISCUSSION THREAD

**Discussion by sebj54** [directus/directus#8618] [Archive]
Hi everyone!

I am actually facing a permission issue in my current project. It is allowed to create a file but read access is restricted given on a custom `directus_files` field (`is_public`).

![image](https://user-images.githubusercontent.com/5880435/136418768-9ab9ac5c-ce7d-4e99-b17e-768f22a69a63.png)

![image](https://user-images.githubusercontent.com/5880435/136418794-78a955d4-6245-4b98-aaac-6bdf21747724.png)

Upload works as expected (`POST /files`) but the API response is empty if the file cannot be read. The problem is that it is impossible to get the ID of the uploaded file. I presume it happens with any collection so maybe someone has a solution for this problem?

I tried to change the response in a `files.create` hook (I wanted to manually add the new file ID) but it is not possible to access the response in a hook.

The only suggestion would be to upload files from a custom endpoint? :/

Thanks in advance for your help!

**Comment by rijkvanzanten** [comment]
The response on create is based on the read permissions. Does this user have the read permissions for the id field?

**Comment by sebj54** [comment]
No that's part of the problem. It is on purpose because the point here is only to upload something then get the file ID to create another item on another collection. The file is only read by an admin.

**Reply by alexvdvalk** [reply]
Very late reply here. You can provide read only access to the ID field only. Open the permissions, then select "Use custom" under your desired collection. There you will be able to select which fields your users can read.

**Reply by johnkmzhou** [reply]
That won't prevent someone from just doing a `GET /files`, taking each id, and appending it: `/assets/:id`.

**Reply by sebj54** [reply]
If I remember well, I ended up with a new custom endpoint which creates the record as an admin and then returns the ID

**Reply by SeanDylanGoff** [reply]
linking for discoverability: https://github.com/directus/directus/discussions/11464 would (IMO) solve this

**Reply by rijkvanzanten** [reply]
@SeanDylanGoff Fully agree
